### DISCUSSION TITLE
Streamlink isn't detecting VLC despite it being in the correct place

### DISCUSSION THREAD

**Discussion by ghost** [streamlink/streamlink#3899] [Q&A] [answered]
VLC is in "C:\Program Files\VideoLAN\VLC\vlc.exe". This is what happens when I try to open a stream:

streamlink https://www.youtube.com/watch?v=tvMXVVYRTnQ&pp=sAQA best
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=tvMXVVYRTnQ
Available streams: 144p (worst), 240p, 360p, 480p, 720p (best)
'pp' is not recognized as an internal or external command,
operable program or batch file.

How can I get it working?

**Comment by bastimeyer** [comment, marked as answer]
> 'pp' is not recognized as an internal or external command, operable program or batch file.

This has nothing to do with VLC, or even Streamlink.

You are passing `youtube.com/watch?v=tvMXVVYRTnQ&pp=sAQA` as an argument to the `streamlink` command while using the Windows command line shell. This shell uses the Windows batch script syntax and interprets the ampersand character `&` as a special character and splits the command at this position. The following string beginning with `pp` is not a valid command, hence the error message.

You'll have to escape special characters or put entire arguments which contain special characters in quotation marks.

Or just google the error message.
https://stackoverflow.com/questions/1327431/how-do-i-escape-ampersands-in-batch-files
